olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -371,6 +358,53 @@ bool DefaultCacheImpl::PromoteKeyLru(const std::string& key) {
   return true;
 }
 
+uint64_t DefaultCacheImpl::MaybeEvictData(leveldb::WriteBatch& batch) {
+  if (!mutable_cache_ || !mutable_cache_lru_) {
+    return 0;
+  }
+
+  const auto max_size = kMaxDiskUsedThreshold * settings_.max_disk_storage;
+  if (mutable_cache_data_size_ < max_size) {
+    return 0;
+  }
+
+  const auto start = std::chrono::steady_clock::now();
+  int64_t evicted = 0u;
+  auto count = 0u;
+  const auto min_size = kMinDiskUsedThreshold * settings_.max_disk_storage;
+  for (auto it = mutable_cache_lru_->rbegin();
+       it != mutable_cache_lru_->rend() &&
+       mutable_cache_data_size_ - evicted > min_size;) {
+    const auto& key = it->key();
+    const auto expiry_key = CreateExpiryKey(key);
+    auto expiry_value = mutable_cache_->Get(expiry_key);
+    if (expiry_value) {
+      evicted += expiry_key.size() + kExpiryValueSize;
+      batch.Delete(expiry_key);
+    }
+    evicted += key.size() + it->value();
+    ++count;
+
+    if (memory_cache_) {
+      memory_cache_->Remove(it->key());
+    }
+
+    batch.Delete(key);
+    mutable_cache_lru_->Erase(it);
+    it = mutable_cache_lru_->rbegin();
```

Comment:
mutable_cache_lru->Erase() returns the next iterator:

```cpp
const_iterator Erase(const_iterator& it) {
    auto prev = it++;
    Erase(prev->key());
    return it;
  }
```

so you should be able to do:

```cpp
it = mutable_cache_lru_->Erase(it);
```

Reply:
well yes, but actually no :-)
The problem here - lru cache has no reverse_iterator thing.
We are removing least used - means the last element. Next element after the last one is mutable_cache_lru_->End(), not new 'last' element.

Reply:
Ok